### PR TITLE
Fix mutex

### PR DESCRIPTION
**Pull request by heavenboy8**

### CODE REVIEW

File: crate/agent/src/conf.rs [hunk 1]
```diff
@@ -46,43 +50,40 @@ pub struct App {
...
+        Self::_relative_to_data_storage(&self.agent.data_storage, &self.agent.ssl_private_key)
+    }
 
-#[derive(Deserialize, Serialize)]
-pub enum EncryptedAppConfAlgorithm {
-    #[serde(rename = "aes256-gcm")]
-    Aes256Gcm,
+    pub fn app_storage(&self) -> Option<PathBuf> {
+        self.app
+            .clone()
```
Comment (ThibsG) [inline crate/agent/src/conf.rs:77]:
```suggestion
            .as_ref()
```

File: resources/conf/agent.toml [hunk 2]
```diff
@@ -1,12 +1,33 @@
...
 port = 5355
-ssl_certificate = "/etc/cosmian_vm/cert.pem"
-ssl_private_key = "/etc/cosmian_vm/key.pem"
+# The location of the SSL certificate used to communicate with the `cosmian_vm_agent`
+# It is recommanded to put that directory inside the directory defined in the `data_storage` field
+# If the path is relative, the location will be searched in the `data_storage` value
+ssl_certificate = "data/cert.pem"
+# The location of the SSL private key used to communicate with the `cosmian_vm_agent`
+# It is recommanded to put that directory inside the directory defined in the `data_storage` field
+# If the path is relative, the location will be searched in the `data_storage` value
```
Comment (ThibsG) [inline resources/conf/agent.toml:19]:
```suggestion
# It is recommended to put that directory inside the directory defined in the `data_storage` field
# If the path is relative, the location path will be joined with the `data_storage` value
ssl_certificate = "data/cert.pem"
# The location of the SSL private key used to communicate with the `cosmian_vm_agent`
# It is recommended to put that directory inside the directory defined by the `data_storage` field
# If the path is relative, the location path will be joined with the `data_storage` value
```

File: resources/conf/agent.toml [hunk 3]
```diff
@@ -1,12 +1,33 @@
...
+ssl_private_key = "data/key.pem"
+# The TPM device path
 tpm_device = "/dev/tpmrm0"
 
 [app]
+# The service type monitoring the user application
+# It could be: supervisor|systemd|standalone
```
Comment (ThibsG) [inline resources/conf/agent.toml:26]:
```suggestion
# Possible values: supervisor|systemd|standalone
```

File: resources/conf/agent.toml [hunk 4]
```diff
@@ -1,12 +1,33 @@
...
 service_type = "supervisor"
+# The name of the service monitoring the use application
```
Comment (ThibsG) [inline resources/conf/agent.toml:28]:
```suggestion
# The name of the service monitoring the used application
```

File: resources/conf/agent.toml [hunk 5]
```diff
@@ -1,12 +1,33 @@
...
 service_app_name = "cosmian_helloworld"
-decrypted_folder = "/mnt/cosmian_vm/data"
-encrypted_secret_app_conf = "/etc/cosmian_vm/app_secrets.json"
+# The location of the user application data
+# It is recommanded to put that directory inside the directory defined in the `data_storage` field
+# If the path is relative, the location will be searched in the `data_storage` value
```
Comment (ThibsG) [inline resources/conf/agent.toml:32]:
```suggestion
# The location of the user application data
# It is recommended to put that directory inside the directory defined by the `data_storage` field
# If the path is relative, the location path will be joined with the `data_storage` value
```

File: resources/conf/agent_no_tpm.toml [hunk 6]
```diff
@@ -1,11 +1,31 @@
...
+# Cosmian VM data storage
+# It contains:
+# - `data/`: a mounted luks container (for persistent data)
+# - `tmp/`: a mounted encrypted tmpfs (for volatile data)
+# - `container`: the luks container
+# Cosmian VM agent will create `data` and `container` at its first start 
+data_storage = "/var/lib/cosmian_vm/"
+# The `cosmian_vm_agent` listens on that host 
 host = "127.0.0.1"
+# The `cosmian_vm_agent` listens on that port
```
Comment (ThibsG) [inline resources/conf/agent_no_tpm.toml:11]:
```suggestion
# The `cosmian_vm_agent` listens to that host 
host = "127.0.0.1"
# The `cosmian_vm_agent` listens to that port 
```

File: resources/conf/agent_no_tpm.toml [hunk 7]
```diff
@@ -1,11 +1,31 @@
...
 port = 5355
-ssl_certificate = "/etc/cosmian_vm/cert.pem"
-ssl_private_key = "/etc/cosmian_vm/key.pem"
+# The location of the SSL certificate used to communicate with the `cosmian_vm_agent`
+# It is recommanded to put that directory inside the directory defined in the `data_storage` field
+# If the path is relative, the location will be searched in the `data_storage` value
+ssl_certificate = "data/cert.pem"
+# The location of the SSL private key used to communicate with the `cosmian_vm_agent`
+# It is recommanded to put that directory inside the directory defined in the `data_storage` field
+# If the path is relative, the location will be searched in the `data_storage` value
```
Comment (ThibsG) [inline resources/conf/agent_no_tpm.toml:19]:
```suggestion
# The location of the SSL certificate used to communicate with the `cosmian_vm_agent`
# It is recommended to put that directory inside the directory defined by the `data_storage` field
# If the path is relative, the location path will be joined with the `data_storage` value
ssl_certificate = "data/cert.pem"
# The location of the SSL private key used to communicate with the `cosmian_vm_agent`
# It is recommended to put that directory inside the directory defined by the `data_storage` field
# If the path is relative, the location path will be joined with the `data_storage` value
```

File: resources/conf/agent_no_tpm.toml [hunk 8]
```diff
@@ -1,11 +1,31 @@
...
+# It could be: supervisor|systemd|standalone
 service_type = "supervisor"
-service_app_name = "cosmian_helloworld"
-decrypted_folder = "/mnt/cosmian_vm/data"
-encrypted_secret_app_conf = "/etc/cosmian_vm/app_secrets.json"
+# The name of the service monitoring the use application
+service_name = "cosmian_helloworld"
+# The location of the user application data
+# It is recommanded to put that directory inside the directory defined in the `data_storage` field
+# If the path is relative, the location will be searched in the `data_storage` value
```
Comment (ThibsG) [inline resources/conf/agent_no_tpm.toml:30]:
```suggestion
# The location of the user application data
# It is recommended to put that directory inside the directory defined by the `data_storage` field
# If the path is relative, the location path will be joined with the `data_storage` value
```

File: resources/scripts/cosmian_fstool [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+if [ -e "$CONTAINER_PATH" ]; then
+    echo "A container already exists in $CONTAINER_PATH"
+    exit 1
+fi
+
+# Make sure to close/umount existing container
+if [ -e "$CONTAINER_MAPPING_PATH" ]; then
+    echo "Closing previous mounted container..."
+    umount "$CONTAINER_MAPPING_PATH" || true
+    cryptsetup close "$CONTAINER_MAPPING_NAME" || true
```
Comment (ThibsG) [inline resources/scripts/cosmian_fstool:115]: what do you think about running `sync` command to be sure?

File: resources/scripts/cosmian_fstool [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+# Encrypt the container (a password is required to run this command)
+echo "Encrypting the container (with password=$PASSWORD)..."
+echo -n "$PASSWORD" | cryptsetup luksFormat "$CONTAINER_PATH" -d -
+
+# Make sure to close/umount existing container
+if [ -e "$CONTAINER_MAPPING_PATH" ]; then
+    echo "Closing previous mounted container..."
+    umount "$CONTAINER_MAPPING_PATH" || true
+    cryptsetup close "$CONTAINER_MAPPING_NAME" || true
```
Comment (ThibsG) [inline resources/scripts/cosmian_fstool:130]: same here, what do you think about running `sync` command to be sure?

**Review by ThibsG — COMMENTED** [review]
Thanks!

File: resources/scripts/cosmian_fstool (hunk 9 above)
Comment (heavenboy8) [reply resources/scripts/cosmian_fstool:115]: ok

**Review by ThibsG — APPROVED** [review]

**Pull request merged**